### ISSUE TITLE
SC2236 not reported on MacOS, but is reported on debian9

### ISSUE THREAD

**Issue by butchhoward** [koalaman/shellcheck#1328]
#### For bugs (or maybe a confusion on my part)
- Rule Id (if any, e.g. SC2236): 
- My shellcheck version (`shellcheck --version` or "online"): 

##### version on debain 9 (installed using wget/extract to get 0.5.0)
ShellCheck - shell script analysis tool
version: 0.5.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

##### version on MaxOS (installed with brew)
ShellCheck - shell script analysis tool
version: 0.5.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2236
- [x] I tried on shellcheck.net and verified that the error is reported as desired (??)
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
CURRENT_ID=''

if [ ! -z "${CURRENT_ID}" ]; then 
        echo "foo"
fi
```

#### Here's what shellcheck currently says:
On debian9
if [ ! -z "${CURRENT_ID}" ]; then 
      ^-- SC2236: Use -n instead of ! -z.

on MacOS
-- no error

#### Here's what I wanted or expected to see:
The same on both regardless (i.e. flag the use of `! -z`)

**Comment by ngzhian** [comment]
Hm, this is a bit weird, but i think the debian version is newer than the macos version you have.
This check was added in https://github.com/koalaman/shellcheck/commit/3d03b0ab3b818ccf51bc38fa7ca19dc2475c17f6 which is not in 0.5.0 yet (this is 08-29), but 0.5.0 is in May (https://github.com/koalaman/shellcheck/releases).

**Comment by koalaman** [comment]
When you download a freshly built binary now, `--version` will still say `0.5.0` and that's what's happened here. This is because I haven't figured out a good way to make `--version` say `0.5.0+3d03b0ab` or `0.5.0+git` or something like that. 

If you download the stable release, `--version` will still say 0.5.0 but it will not include this check.
